client/scala/src/main/scala/ai/verta/repository/Commit.scala

```diff
@@ -333,9 +346,24 @@ class Commit(
    *  @return the new Commit instance, if succeeds.
    */
   def applyDiff(diff: Diff, message: String)(implicit ec: ExecutionContext) = {
+    checkCommit("Commit must be saved before a diff can be applied").flatMap(_ =>
+      createCommit(message = message, diffs = diff.blobDiffs, commitBase = id)
+    )
+  }
+
+  /** Check that the commit is saved. If other commit is passed, check that it is saved, and both are in the same repository
+   *  @param message error message if this commit is not saved
+   *  @param other (optional) the other commit
+   *  @param otherMessage (optional) the error message if the other commit is not saved
+   */
+  private def checkCommit(message: String, other: Option[Commit] = None, otherMessage: Option[String] = None): Try[Unit] = {
```

Comment:
This method is pretty complex and trying to cover a particular case. I'd recommend building a base functionality and seeing if you can re-use that component instead of trying to cover everything possible in a single method. Composition is key.

Reply:
Hmm how about split this into two methods: one for checking saved state, and one for checking if two commits belogn to the same repository?

Reply:
I just separate it into checkSaved and checkSameRepository. See if it looks better.